OpenEdXMobile/src/main/java/org/edx/mobile/util/links/DefaultActionListener.java

```diff
@@ -162,7 +162,10 @@ public void onClickEnroll(@NonNull final String courseId, final boolean emailOpt
                     protected void onResponse(@NonNull final ResponseBody responseBody) {
                         super.onResponse(responseBody);
                         logger.debug("Enrollment successful: " + courseId);
-                        Toast.makeText(activity, activity.getString(R.string.you_are_now_enrolled), Toast.LENGTH_SHORT).show();
+                        Toast.makeText(activity, activity.getString(
+                                courseApi.isCourseEnrolled(courseId) ?
```

Comment:
it's better to check if the course is already enrolled before the service call of course enrollment and move to the course dashboard without calling an extra service.